src/endpoint/s3/ops/s3_put_bucket_lifecycle.js

```diff
@@ -171,10 +171,19 @@ async function put_bucket_lifecycle(req) {
         return current_rule;
     });
 
-    await req.object_sdk.set_bucket_lifecycle_configuration_rules({
-        name: req.params.bucket,
-        rules: lifecycle_rules
-    });
+    try {
+        await req.object_sdk.set_bucket_lifecycle_configuration_rules({
+            name: req.params.bucket,
+            rules: lifecycle_rules
+        });
+    } catch (error) {
+        let err = error;
+        if (error.rpc_code === "INVALID_SCHEMA" || error.rpc_code === "INVALID_SCHEMA_PARAMS") {
+            err = new S3Error(S3Error.MalformedXML);
+            err.message = "The XML you provided was not well-formed or did not validate against our published schema";
+        }
+        throw err;
+    }
```

Comment:
I prefer "Fail First" concept ([reference](https://stackoverflow.com/questions/2807241/what-does-the-expression-fail-early-mean-and-when-would-you-want-to-do-so)).
Did you try to go over the rules and validate their status value?
In the [AWS S3 documentation](https://docs.aws.amazon.com/AmazonS3/latest/API/API_LifecycleRule.html), I can see that we have 2 options only ("Valid Values: Enabled | Disabled").

Reply:
Yes, you are correct. I am failing it at the time of validation. But I think we can add still return a different error from here just like [this](https://github.com/noobaa/noobaa-core/blob/0f3d341bdb36c6a22db401da7d43696e2a22647f/src/endpoint/s3/ops/s3_put_bucket_policy.js#L17C5-L27C2)

Reply:
I'm suggesting to:
1. Validate the status in the rules before it passes inside.
2. Fix the message of `MalformedXML` to be the actual message and move the text before as a comment (so we will not need to rewrite it every time).
3. Change the translation you copied as a function that will be reused
WDYT?

Reply:
Updated, Thanks